diff --git a/src/openvpn/common.h b/src/openvpn/common.h
index 3a84541c..390abe92 100644
--- a/src/openvpn/common.h
+++ b/src/openvpn/common.h
@@ -66,7 +66,7 @@ typedef unsigned long ptr_type;
  * maximum size of a single TLS message (cleartext).
  * This parameter must be >= PUSH_BUNDLE_SIZE
  */
-#define TLS_CHANNEL_BUF_SIZE 2048
+#define TLS_CHANNEL_BUF_SIZE 65536
 
 /* TLS control buffer minimum size
  *
diff --git a/src/openvpn/manage.c b/src/openvpn/manage.c
index feb32274..d4c997f2 100644
--- a/src/openvpn/manage.c
+++ b/src/openvpn/manage.c
@@ -2245,7 +2245,7 @@ man_read(struct management *man)
     /*
      * read command line from socket
      */
-    unsigned char buf[256];
+    unsigned char buf[TLS_CHANNEL_BUF_SIZE];
     int len = 0;
 
 #ifdef TARGET_ANDROID
@@ -2581,7 +2581,7 @@ man_connection_init(struct management *man)
          * Allocate helper objects for command line input and
          * command output from/to the socket.
          */
-        man->connection.in = command_line_new(1024);
+        man->connection.in = command_line_new(TLS_CHANNEL_BUF_SIZE);
         man->connection.out = buffer_list_new();
 
         /*
diff --git a/src/openvpn/misc.h b/src/openvpn/misc.h
index 24a1ca56..6d41d6d7 100644
--- a/src/openvpn/misc.h
+++ b/src/openvpn/misc.h
@@ -65,7 +65,7 @@ struct user_pass
 #ifdef ENABLE_PKCS11
 #define USER_PASS_LEN 4096
 #else
-#define USER_PASS_LEN 128
+#define USER_PASS_LEN 65536
 #endif
     /* Note that username and password are expected to be null-terminated */
     char username[USER_PASS_LEN];
diff --git a/src/openvpn/options.h b/src/openvpn/options.h
index e85d8062..9d2f743e 100644
--- a/src/openvpn/options.h
+++ b/src/openvpn/options.h
@@ -54,8 +54,8 @@
 /*
  * Max size of options line and parameter.
  */
-#define OPTION_PARM_SIZE 256
-#define OPTION_LINE_SIZE 256
+#define OPTION_PARM_SIZE USER_PASS_LEN
+#define OPTION_LINE_SIZE USER_PASS_LEN
 
 extern const char title_string[];
 
